fv3core/stencils/saturation_adjustment.py

```diff
@@ -886,103 +885,105 @@ def satadjust(
             pkz = compute_pkz_func(dp, delz, pt, cappa)
 
 
-def compute(
-    te: FloatField,
-    qvapor: FloatField,
-    qliquid: FloatField,
-    qice: FloatField,
-    qrain: FloatField,
-    qsnow: FloatField,
-    qgraupel: FloatField,
-    qcld: FloatField,
-    hs: FloatFieldIJ,
-    peln: FloatField,
-    delp: FloatField,
-    delz: FloatField,
-    q_con: FloatField,
-    pt: FloatField,
-    pkz: FloatField,
-    cappa: FloatField,
-    r_vir: float,
-    mdt: float,
-    fast_mp_consv: bool,
-    last_step: bool,
-    akap: float,
-    kmp: int,
-) -> None:
-
-    grid = spec.grid
-    origin = (grid.is_, grid.js, kmp)
-    domain = (grid.nic, grid.njc, (grid.npz - kmp))
-    hydrostatic = spec.namelist.hydrostatic
-    sdt = 0.5 * mdt  # half remapping time step
-    # define conversion scalar / factor
-    fac_i2s = 1.0 - math.exp(-mdt / spec.namelist.tau_i2s)
-    fac_v2l = 1.0 - math.exp(-sdt / spec.namelist.tau_v2l)
-    fac_r2g = 1.0 - math.exp(-mdt / spec.namelist.tau_r2g)
-    fac_l2r = 1.0 - math.exp(-mdt / spec.namelist.tau_l2r)
-
-    fac_l2v = 1.0 - math.exp(-sdt / spec.namelist.tau_l2v)
-    fac_l2v = min(spec.namelist.sat_adj0, fac_l2v)
-
-    fac_imlt = 1.0 - math.exp(-sdt / spec.namelist.tau_imlt)
-    fac_smlt = 1.0 - math.exp(-mdt / spec.namelist.tau_smlt)
```

Comment:
I want to double check with folks, on whether we want these calculations to happen inside of stencils. in theory, yes we do, so probably fine to do. but if there's any lag in toolchain. logic that accidentally makes 3d fields out of these scalars, we might want to reconsider.